### PR TITLE
Improving dob

### PR DESCRIPTION
**Pull request by Juxarius**
Small clean up on sort.

Improved DOB format
- Now dob are stored as `LocalDate` format with the Java8 update
- dob parser is smart to allow single digits for date and month, 2 or 4 digits for year
- also allows month name to be entered instead
- however is restricted to day-month-year

please merge asap

### CODE REVIEW

File: src/main/java/seedu/address/model/person/DateOfBirth.java [hunk 1]
```diff
@@ -43,7 +58,82 @@ public DateOfBirth(String dob) throws IllegalValueException {
...
+
+    /**
+     * Parses input dob string
+     */
+    private LocalDate dateFormatter(String dob) throws IllegalValueException {
+        List<String> arguments = Arrays.asList(dob.split("[\\s-/.]"));
+        String day = arguments.get(0);
+        String month = arguments.get(1);
+        String year = arguments.size() > 2 ? arguments.get(2) : String.valueOf(LocalDate.now().getYear());
+        if (arguments.size() > 3 || arguments.size() < 2) {
```
Comment (OscarWang114) [inline src/main/java/seedu/address/model/person/DateOfBirth.java]: It'd be more efficient if we validate the argument first. Can move line 73 to the front.

**Review by OscarWang114 — CHANGES_REQUESTED** [review]
One suggestion. Will merge after successfully pacifying Travis.

File: src/main/java/seedu/address/model/person/DateOfBirth.java (hunk 1 above)
Comment (Juxarius) [reply src/main/java/seedu/address/model/person/DateOfBirth.java]: done~

**Review by OscarWang114 — COMMENTED** [review]
LGTM! 
Mergeing after passing Travis

**Review by OscarWang114 — APPROVED** [review]
Approved

**Pull request merged**